distribution/push_v2.go

```diff
@@ -670,10 +668,9 @@ func numOfMatchingPathComponents(pth string, matchComponents []string) int {
 }
 
 func getPathComponents(path string) []string {
-	// make sure to add docker.io/ prefix to the path
-	named, err := reference.ParseNamed(path)
+	named, err := reference.ParseNormalizedNamed(path)
```

Comment:
Pushed changes, haven't resolved this one yet, not quite sure what the best route is. There may be something incorrect about `SourceRepository` right now though, I am not sure it is always normalized.  If it is always normalized I can just make this split and nothing else.